python/cugraph/cugraph/utilities/api_tools.py

```diff
@@ -11,83 +11,29 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-import functools
-import warnings
-import inspect
-import types
+import pylibcugraph.utilities.api_tools as api_tools
 
 experimental_prefix = "EXPERIMENTAL"
 
 
-# FIXME: this utility is copied from pylibcugraph. Remove this copy and have
-# cugraph code call the version in pylibcugraph.
-
 def experimental_warning_wrapper(obj):
     """
     Wrap obj in a function or class that prints a warning about it being
     "experimental" (ie. it is in the public API but subject to change or
     removal), prior to calling obj and returning its value.
-
     The object's name used in the warning message also has any leading __
     and/or EXPERIMENTAL string are removed from the name used in warning
     messages. This allows an object to be named with a "private" name in the
     public API so it can remain hidden while it is still experimental, but
     have a public name within the experimental namespace so it can be easily
     discovered and used.
     """
-    obj_type = type(obj)
-    if obj_type not in [type, types.FunctionType, types.BuiltinFunctionType]:
-        raise TypeError("obj must be a class or a function type, got "
-                        f"{obj_type}")
-
-    obj_name = obj.__name__
-    obj_name = obj_name.lstrip(experimental_prefix)
-    obj_name = obj_name.lstrip("__")
-
-    # Assume the caller of this function is the module containing the
-    # experimental obj and try to get its namespace name. Default to no
-    # namespace name if it could not be found.
-    call_stack = inspect.stack()
-    calling_frame = call_stack[1].frame
-    ns_name = calling_frame.f_locals.get("__name__")
-    dot = "." if ns_name is not None else ""
-
-    warning_msg = (f"{ns_name}{dot}{obj_name} is experimental and will "
-                   "change or be removed in a future release.")
+    return api_tools.experimental_warning_wrapper(obj)
```

Comment:
I like this

Reply:
To follow the same logic as `promoted_experimental_warning_wrapper()` and `def deprecated_warning_wrapper()`, is it not better to get rid of `experimental_warning_wrapper` docstrings and just have `def experimental_warning_wrapper()`

Reply:
Good point, done.